### ISSUE TITLE
Marker for x-kubernetes-preserve-unknown-fields

### ISSUE THREAD

**Issue by hasheddan** [kubernetes-sigs/controller-tools#281]
Similar to #278, a marker should be created to specify pruning behavior as was introduced in [v1.15](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#pruning-versus-preserving-unknown-fields). There would also need to be an integration test case for this marker.

Proposal for format of marker: `+kubebuilder:validation:PreserveUnknownFields`

**Comment by DirectXMan12** [comment]
seems reasonable to me :-)

**Comment by liggitt** [comment]
note that `spec.preserveUnknownFields: true` will not be allowed in v1, but must be specified in the schema itself, for example:
```yaml
versions:
- name: v1
  schema:
    type: "object"
    x-kubernetes-preserve-unknown-fields: true
- name: v2
  schema:
    type: "object"
    properties:
      myproperty:
        type: "object"
        x-kubernetes-preserve-unknown-fields: true
```

Also note that use of `x-kubernetes-preserve-unknown-fields: true` to avoid specifying any schema properties is discouraged, since it limits the amount of useful information available to clients. Use for specific subtrees that can contain arbitrary JSON is the intended use case.

**Comment by DirectXMan12** [comment]
yep, the marker would set the x- value.

**Comment by hasheddan** [comment]
Fixed by #345 

/close
